src/volatile_memory.rs

```diff
@@ -181,7 +189,9 @@ pub trait VolatileMemory {
         Ok(&*(slice.addr as *const T))
     }
 
-    /// Returns a mutable reference to an instance of `T` at `offset`.
+    /// Returns a mutable reference to an instance of `T` at `offset`. Mutable accesses performed
```

Comment:
Can we add the same comment about the mutable accesses not being marked as dirty to `as_ptr()` documentation as well?